src/applications/disability-benefits/all-claims/containers/ITFWrapper.jsx

```diff
@@ -89,73 +67,51 @@ export class ITFWrapper extends React.Component {
   }
 
   render() {
-    const { location, children, itf } = this.props;
-    // If the location is the intro or confirmation pages, don't show an ITF message
-    let message;
-    let content;
+    const { itf } = this.props;
 
-    if (this.shouldBlockITF(location.pathname)) {
-      message = null;
-      content = children;
+    if (this.shouldBlockITF(this.props.location.pathname)) {
+      return this.props.children;
     } else if (fetchWaitingStates.includes(itf.fetchCallState)) {
       // If we get here, componentDidMount or componentWillRecieveProps called fetchITF
       // While we're waiting, show the loading indicator...
-      content = (
+      return (
         <LoadingIndicator message="Please wait while we check your ITF status." />
       );
     } else if (itf.fetchCallState === requestStates.failed) {
       // We'll get here after the fetchITF promise is fulfilled
-      message = itfMessage(
-        'We’re sorry. Something went wrong on our end.',
-        itfError,
-        'error',
-      );
+      return <ITFBanner status="error" />;
     } else if (itf.currentITF && itf.currentITF.status === itfStatuses.active) {
-      // If we have an active ITF, we're good to go--render that form!
-      content = children;
-
-      // Only show the message on the first page navigation
-      if (!this.state.hasDisplayedSuccess) {
-        const { expirationDate: currentExpDate } = itf.currentITF;
-        if (itf.previousITF) {
-          const { expirationDate: prevExpDate } = itf.previousITF;
-          // If there was a previous ITF, we created one; show the creation success message
-          message = itfMessage(
-            'Your Intent to File request has been submitted',
-            itfSuccess(itf.previousITF, currentExpDate, prevExpDate),
-            'success',
-          );
-        } else {
-          // We fetched an active ITF
-          message = itfMessage(
-            'You already have an Intent to File',
-            itfActive(currentExpDate),
-            'success',
-          );
-        }
+      const { expirationDate: currentExpDate } = itf.currentITF;
+      if (itf.previousITF) {
+        const { expirationDate: prevExpDate } = itf.previousITF;
+        // If there was a previous ITF, we created one; show the creation success message
+        return (
+          <ITFBanner
```

Comment:
If you wanted to avoid having to pass all these props, you could use a render prop and keep more of the logic in this component.